### PR TITLE
Add nbproject directory to .gitignore

### PR DESCRIPTION
**Pull request by odonnellnoel**
Hi, this is my first patch.

I'm [one of the few] using NetBeans, and this PR adds the `nbproject` directory created by NetBeans to `.gitignore`.

Hope to make more contributions soon!

### CODE REVIEW

File: external/db_drivers/liblmdb/mdb.c [hunk 1]
```diff
@@ -176,6 +176,7 @@ typedef SSIZE_T	ssize_t;
 #include <semaphore.h>
 #elif defined(MDB_USE_SYSV_SEM)
 #include <sys/ipc.h>
+#define _WANT_SEMUN             1
```
Comment (vtnerd) [inline external/db_drivers/liblmdb/mdb.c]: This change seems unrelated ... ?

Comment (odonnellnoel) [reply external/db_drivers/liblmdb/mdb.c]: Sorry, I had no idea it would include subsequent commits in the PR. I've `git reset --force`'d away the last commit.

**Comment by odonnellnoel** [comment]
Could somebody please review this PR?
